### PR TITLE
BPMSPL 781

### PR DESCRIPTION
**Pull request by mcivantos-tribalyte**
Allow to search tasks assigned by potential owner through Advanced Queries

### CODE REVIEW

File: jbpm-services/jbpm-services-api/src/main/java/org/jbpm/services/api/query/QueryResultMapper.java [hunk 1]
```diff
@@ -98,6 +99,17 @@
     public static final String COLUMN_WORKITEMID = "WORKITEMID";
     public static final String COLUMN_ORGANIZATIONAL_ENTITY = "ID";
     public static final String COLUMN_EXCLUDED_OWNER = "ENTITY_ID";
+    public static final String COLUMN_EXPIRATIONTIME = "EXPIRATIONTIME";
+    public static final String COLUMN_POTOWNER = "POTOWNER";
+    public static final String COLUMN_TASK_TYPE = "TASKTYPE";
+    public static final String COLUMN_FORM_NAME = "FORMNAME";
+    public static final String COLUMN_SUMMARY = "SUMMARY";
+    public static final String COLUMN_OPTLOCK = "OPTLOCK";
```
Comment (mswiderski) [inline jbpm-services/jbpm-services-api/src/main/java/org/jbpm/services/api/query/QueryResultMapper.java]: potluck is an internal hibernate thing so should not really be returned as part of the queries.

File: jbpm-services/jbpm-services-api/src/main/java/org/jbpm/services/api/model/UserTaskInstanceWithPotOwnerDesc.java [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+ * limitations under the License.
+ */
+
+package org.jbpm.services.api.model;
+
+import java.util.Date;
+import java.util.List;
+import java.util.Map;
+
+public interface UserTaskInstanceWithPotOwnerDesc {
```
Comment (mswiderski) [inline jbpm-services/jbpm-services-api/src/main/java/org/jbpm/services/api/model/UserTaskInstanceWithPotOwnerDesc.java]: why not extending UserTaskInstanceDesc and add only additional methods?

File: jbpm-services/jbpm-kie-services/src/main/java/org/jbpm/kie/services/impl/query/mapper/UserTaskInstanceWithPotOwnerQueryMapper.java [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+            
+            return mappedResult;
+        }
+        
+        throw new IllegalArgumentException("Unsupported result for mapping " + result);
+    }
+
+    @Override
+    public String getName() {
+        return "CustomUserTasks";
```
Comment (mswiderski) [inline jbpm-services/jbpm-kie-services/src/main/java/org/jbpm/kie/services/impl/query/mapper/UserTaskInstanceWithPotOwnerQueryMapper.java]: it's not really CustomUserTasks but rather UserTasksWithPotOwners

File: jbpm-services/jbpm-kie-services/src/main/java/org/jbpm/kie/services/impl/query/mapper/UserTaskInstanceWithModifVarsQueryMapper.java [hunk 4]
```diff
@@ -0,0 +1,116 @@
+package org.jbpm.kie.services.impl.query.mapper;
```
Comment (mswiderski) [inline jbpm-services/jbpm-kie-services/src/main/java/org/jbpm/kie/services/impl/query/mapper/UserTaskInstanceWithModifVarsQueryMapper.java]: missing license header

File: jbpm-services/jbpm-kie-services/src/main/java/org/jbpm/kie/services/impl/model/UserTaskInstanceWithPotOwnerDesc.java [hunk 5]
```diff
@@ -0,0 +1,332 @@
...
+
+import java.io.Serializable;
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+
+public class UserTaskInstanceWithPotOwnerDesc implements org.jbpm.services.api.model.UserTaskInstanceWithPotOwnerDesc, Serializable {
```
Comment (mswiderski) [inline jbpm-services/jbpm-kie-services/src/main/java/org/jbpm/kie/services/impl/model/UserTaskInstanceWithPotOwnerDesc.java]: same here, why not to extend UserTaskInstanceDesc and add additional fields

File: jbpm-human-task/jbpm-human-task-core/src/main/java/org/jbpm/services/task/identity/CustomUserGroupCallbackImpl.java [hunk 6]
```diff
@@ -0,0 +1,152 @@
...
+
+import org.kie.api.task.UserGroupCallback;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+
+/**
+ * Custom User Group Callback for Human Tasks
+ */
+public class CustomUserGroupCallbackImpl implements UserGroupCallback {
```
Comment (mswiderski) [inline jbpm-human-task/jbpm-human-task-core/src/main/java/org/jbpm/services/task/identity/CustomUserGroupCallbackImpl.java]: in general this callback has very limited value and frankly speaking no value for production deployments, note that application-users.properties and application-roles.properties are EAP/WidFly specific and we already have JBoss like user group callback - https://github.com/kiegroup/jbpm/blob/master/jbpm-human-task/jbpm-human-task-core/src/main/java/org/jbpm/services/task/identity/JBossUserGroupCallbackImpl.java.
So I opt at removing this one completely
Here you can find all callbacks we ship out of the box https://github.com/kiegroup/jbpm/blob/master/jbpm-human-task/jbpm-human-task-core/src/main/java/org/jbpm/services/task/identity

File: jbpm-services/jbpm-kie-services/src/main/java/org/jbpm/kie/services/impl/query/builder/UserTaskPotOwnerQueryBuilder.java [hunk 7]
```diff
@@ -0,0 +1,60 @@
...
+    }
+    
+    @Override
+    public ColumnFilter build() {
+     // return null if it was already invoked
+        if (built) {
+            return null;
+        }
+         
+        UserGroupCallback userGroupCallback = new CustomUserGroupCallbackImpl();
```
Comment (mswiderski) [inline jbpm-services/jbpm-kie-services/src/main/java/org/jbpm/kie/services/impl/query/builder/UserTaskPotOwnerQueryBuilder.java]: this is limiting the feature quite a lot, you should use this class instead to use the same callback as other parts of the system (in most of the cases) https://github.com/kiegroup/jbpm/blob/master/jbpm-runtime-manager/src/main/java/org/jbpm/runtime/manager/impl/identity/UserDataServiceProvider.java

File: jbpm-services/jbpm-services-api/src/main/java/org/jbpm/services/api/query/QueryResultMapper.java (hunk 1 above)
Comment (mcivantos-tribalyte) [reply jbpm-services/jbpm-services-api/src/main/java/org/jbpm/services/api/query/QueryResultMapper.java]: Ok Maciej. The customer would like to see the optlock in the queries results. But I remove it and I talk to them.

**Comment by mswiderski** [comment]
and one more thing, please don't use BPMSPL jira as commit message but rather clone the jira into JBPM project as BPMSPL is not visible to all users

File: jbpm-services/jbpm-kie-services/src/main/java/org/jbpm/kie/services/impl/query/mapper/UserTaskInstanceWithPotOwnerQueryMapper.java (hunk 3 above)
Comment (mcivantos-tribalyte) [reply jbpm-services/jbpm-kie-services/src/main/java/org/jbpm/kie/services/impl/query/mapper/UserTaskInstanceWithPotOwnerQueryMapper.java]: Ok!

File: jbpm-services/jbpm-kie-services/src/main/java/org/jbpm/kie/services/impl/query/mapper/UserTaskInstanceWithModifVarsQueryMapper.java (hunk 4 above)
Comment (mcivantos-tribalyte) [reply jbpm-services/jbpm-kie-services/src/main/java/org/jbpm/kie/services/impl/query/mapper/UserTaskInstanceWithModifVarsQueryMapper.java]: Done

File: jbpm-human-task/jbpm-human-task-core/src/main/java/org/jbpm/services/task/identity/CustomUserGroupCallbackImpl.java (hunk 6 above)
Comment (mcivantos-tribalyte) [reply jbpm-human-task/jbpm-human-task-core/src/main/java/org/jbpm/services/task/identity/CustomUserGroupCallbackImpl.java]: Done

File: jbpm-services/jbpm-kie-services/src/main/java/org/jbpm/kie/services/impl/model/UserTaskInstanceWithPotOwnerDesc.java (hunk 5 above)
Comment (mcivantos-tribalyte) [reply jbpm-services/jbpm-kie-services/src/main/java/org/jbpm/kie/services/impl/model/UserTaskInstanceWithPotOwnerDesc.java]: Done

File: jbpm-services/jbpm-kie-services/src/main/java/org/jbpm/kie/services/impl/query/builder/UserTaskPotOwnerQueryBuilder.java (hunk 7 above)
Comment (mcivantos-tribalyte) [reply jbpm-services/jbpm-kie-services/src/main/java/org/jbpm/kie/services/impl/query/builder/UserTaskPotOwnerQueryBuilder.java]: Done

File: jbpm-services/jbpm-services-api/src/main/java/org/jbpm/services/api/model/UserTaskInstanceWithPotOwnerDesc.java (hunk 2 above)
Comment (mcivantos-tribalyte) [reply jbpm-services/jbpm-services-api/src/main/java/org/jbpm/services/api/model/UserTaskInstanceWithPotOwnerDesc.java]: Done

File: jbpm-services/jbpm-services-api/src/main/java/org/jbpm/services/api/query/QueryResultMapper.java (hunk 1 above)
Comment (mcivantos-tribalyte) [reply jbpm-services/jbpm-services-api/src/main/java/org/jbpm/services/api/query/QueryResultMapper.java]: Done.

**Comment by mswiderski** [comment]
@mcivantos-tribalyte same here, wrong branch - it should target master not 7.4.x

**Comment by mswiderski** [comment]
closing as master targeting branch is already opened #1084

**Pull request closed without merging**